packages/rest/test/routes.test.ts

```diff
@@ -441,4 +447,21 @@ describe("rest: routes", () => {
       { verb: "put", path: "/things/{thingId}", params: ["thingId"] },
     ]);
   });
+
+  it("@autoRoute operations can also be shared routes", async () => {
+    const runner = await createRestTestRunner();
+    const { get1, get2 } = (await runner.compile(`
+      @test
+      @autoRoute
+      @sharedRoute
+      op get1(@path name: string): string;
+
+      @test
+      @autoRoute
+      op get2(@path name: string): string;
+    `)) as { get1: Operation; get2: Operation };
```

Comment:
Is it common to actually put `@autoRoute` on individual operations? Why could we not just have `@autoRoute` take the same options object as `@route`? Or just have both make use of this decorator?

Reply:
Yes, this is the pattern used in `Azure.Core`.  The `@autoRoute` decorator is placed on `ResourceOperation<TResource>` which causes it to be applied individually to every operation which uses standard operation signatures via `is`.

In my earlier comment, I explained that we shouldn't conflate sharedness of routes with `@route` or `@autoRoute` because changing an operation signature to go from non-shared to shared means re-applying the original decorator instead of just adding `@sharedRoute` to the operation, no matter what kind of route it uses.  This is the key element of how this PR is intended to affect the design.